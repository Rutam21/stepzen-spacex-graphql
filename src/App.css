.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #19253f;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: #ddd;
  padding: 1rem;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.Title {
  font-size: 22px;
}

.Description {
  font-size: 18px;
}

.RocketsInformation {
  display: block;
  margin-top: 36px;
  background-image: linear-gradient(92.88deg, #5643cc 25%, #9b7af7 75%);
  border-radius: 8px;
  border-style: none;
  box-sizing: border-box;
  color: #ffffff;
  cursor: pointer;
  flex-shrink: 0;
  font-size: 16px;
  font-weight: 500;
  height: 4rem;
  line-height: 4rem;
  padding: 0 5.6rem;
  text-align: center;
  text-shadow: rgba(0, 0, 0, 0.25) 0 3px 8px;
  transition: all 0.5s;
  user-select: none;
  -webkit-user-select: none;
  touch-action: manipulation;
  text-decoration: none;
}

.RocketsInformation:hover {
  box-shadow: rgba(80, 63, 205, 0.5) 0 1px 30px;
  transition-duration: 0.1s;
}
