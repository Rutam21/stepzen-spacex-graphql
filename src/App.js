import "./App.css";
import { gql, useQuery } from "@apollo/client";
import StepZenLogo from "./light-blue.svg";
import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";
import Rockets from "./rocket-information";
import Card from "./Card";

const GET_QUERY = gql`
  query MyQuery {
    capsule(id: "C105") {
      id
      landings
      original_launch
      reuse_count
      missions {
        flight
        name
      }
    }
  }
`;

function App() {
  return (
    <>
      <Router>
        <div>
          <nav>
            <ul>
              {/* <li>
                <Link to='/'>Home</Link>
              </li> */}
              {/* <li>
                <Link to='/rockets'>Rockets Information</Link>
              </li> */}
            </ul>
          </nav>
          <Switch>
            <Route path="/rockets">
              <Rockets />
            </Route>
            <Route path="/">
              <Home />
            </Route>
          </Switch>
        </div>
      </Router>
    </>
  );
}

function Home() {
  const { loading, error, data } = useQuery(GET_QUERY);
  console.log("DATA", data);

  if (loading) return <p>Loading ...</p>;

  if (error) return <pre>{JSON.stringify(error, null, 2)}</pre>;
  return (
    <>
      <Card>
        <div className="App">
          <header className="App-header">
            <img src={StepZenLogo} alt="StepZen Logo" width="200px" />
            <p style={{ marginTop: "40px" }}>Capsule information pulled from StepZen Endpoint:</p>
            <span>id: {data.capsule.id}</span>
            <span>landings: {data.capsule.landings}</span>
            <span>reuse-count: {data.capsule.reuse_count}</span>
            <Link to="/rockets">Rockets Information</Link>
          </header>
        </div>
      </Card>
    </>
  );
}
export default App;
