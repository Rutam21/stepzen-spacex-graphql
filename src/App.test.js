import { render, cleanup } from '@testing-library/react';
import App, { GET_QUERY } from './App';
import { MockedProvider } from '@apollo/client/testing';
import { act } from 'react-dom/test-utils';

afterEach(cleanup);

const dataFixture = {
  capsule: {
    id: '1',
    landings: 0,
    original_launch: {},
    reuse_count: 0,
    missions: [
      {
        flight: 0,
        name: 'test-capsule-mission-name',
      },
    ],
  },
};

const errorMock = {
  request: {
    query: GET_QUERY,
  },
  error: new Error('A test error occurred'),
};

const validMock = {
  request: {
    query: GET_QUERY,
  },
  result: {
    data: dataFixture,
  },
};

test('renders loading screen', () => {
  const { asFragment } = render(
    <MockedProvider mocks={[]} addTypename={false}>
      <App />
    </MockedProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});

test('renders error screen', async () => {
  const { asFragment } = render(
    <MockedProvider mocks={[errorMock]} addTypename={false}>
      <App />
    </MockedProvider>
  );
  await act(async () => {
    await new Promise((resolve) => setTimeout(resolve, 0));
  });
  expect(asFragment()).toMatchSnapshot();
});

test('renders app screen', async () => {
  const { asFragment } = render(
    <MockedProvider mocks={[validMock]} addTypename={false}>
      <App />
    </MockedProvider>
  );
  await act(async () => {
    await new Promise((resolve) => setTimeout(resolve, 0));
  });
  expect(asFragment()).toMatchSnapshot();
});
