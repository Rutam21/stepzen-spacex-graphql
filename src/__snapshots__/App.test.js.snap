// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app screen 1`] = `
<DocumentFragment>
  <div>
    <nav>
      <ul />
    </nav>
    <div
      class="App"
    >
      <header
        class="App-header"
      >
        <img
          alt="StepZen Logo"
          src="light-blue.svg"
          width="200px"
        />
        <p
          style="margin-top: 40px;"
        >
          Capsule information pulled from StepZen Endpoint:
        </p>
        <ul>
          <li>
            id: 1
          </li>
          <li>
            landings: 0
          </li>
          <li>
            reuse-count: 0
          </li>
        </ul>
        <a
          href="/rockets"
        >
          Rockets Information
        </a>
      </header>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders error screen 1`] = `
<DocumentFragment>
  <div>
    <nav>
      <ul />
    </nav>
    <pre>
      {
  "graphQLErrors": [],
  "networkError": {},
  "message": "A test error occurred"
}
    </pre>
  </div>
</DocumentFragment>
`;

exports[`renders loading screen 1`] = `
<DocumentFragment>
  <div>
    <nav>
      <ul />
    </nav>
    <p>
      Loading ...
    </p>
  </div>
</DocumentFragment>
`;
